src/components/KMethod/KMethod.vue

```diff
@@ -0,0 +1,117 @@
+<template>
+  <KBadge
+    :background-color="backgroundColor || methodBadgeColors[method].backgroundColor"
+    class="k-method"
+    :class="`method-${method}`"
+    :color="color || methodBadgeColors[method].color"
+    max-width="auto"
+    :shape="appearance === 'rectangular' ? 'rectangular' : 'rounded'"
+  >
+    {{ methodLabel }}
+  </KBadge>
+</template>
+
+<script lang="ts">
+import { PropType, computed } from 'vue'
+import type { Method, MethodAppearance } from '@/types'
+
+export const methods: Method[] = ['get', 'post', 'put', 'patch', 'delete', 'head', 'options', 'trace', 'connect', 'custom']
+
+export const shapes: MethodAppearance[] = ['rounded', 'rectangular']
+</script>
+
+<script setup lang="ts">
+const props = defineProps({
+  method: {
+    type: String as PropType<Method>,
+    required: true,
+    validator: (value: string): boolean => {
+      return methods.includes(value as Method)
+    },
+    default: 'post',
+  },
+  appearance: {
+    type: String as PropType<MethodAppearance>,
+    required: false,
+    validator: (value: string): boolean => {
+      return shapes.includes(value as MethodAppearance)
+    },
+    default: 'rounded',
+  },
+  label: {
+    type: String,
+    required: false,
+    default: '',
+  },
+  color: {
+    type: String,
+    required: false,
+    default: '',
+  },
+  backgroundColor: {
+    type: String,
+    required: false,
+    default: '',
+  },
+})
+
+const methodLabel = computed((): string => {
+  return props.label || props.method.toUpperCase()
+})
+
+interface MethodBadgeColors {
+  color: string
+  backgroundColor: string
+}
```

Comment:
suggestion: can you move this interface to the types file you created and import from there?